artio-codecs/src/main/java/uk/co/real_logic/artio/util/MutableAsciiBuffer.java

```diff
@@ -617,6 +644,16 @@ private long calculateRemainderAndPutMinus(final int offset, final long value)
         }
     }
 
+    private int checkForLeadingZeroAndPutIfNeeded(final int offset, final long value, final int scale)
```

Comment:
Is this the most efficient way to do this? Can't we just check if the number of zeros is > 0?

Reply:
I tried to minimize the code change.
so i treated the leading zero similar to the minus sign.
if going to refer the zeros count it would lead to larger code change.
why do you think it is not efficient? the access to the array is O(1).

Reply:
Well, given that you had the number of zeros in a local field, It just looked to me you could do a quick check on that. Perhaps I'm miss-reading the diff ;) I don't think your approach is bad, just wondering whether it wasn't going the other route?

Reply:
well I managed to figure it out your way. take a look at new code.

Reply:
Great, thanks.